src/crypto/generic-ops.h

```diff
@@ -34,13 +34,18 @@
 #include <cstring>
 #include <functional>
 
+extern "C"
```

Comment:
I think it makes more sense to do this in the `verify.h` file since this is a primarily a C++ project. `#iifdef __cplusplus` is the usual way of detecting this at the top and bottom of the file.